combat/CombatDamage.cpp

```diff
@@ -0,0 +1,211 @@
+
+        if (target_ships)
+            retval.emplace_back(fighter_damage * fighter_shots);
+        else
+            retval.emplace_back((float)fighter_shots);
+        return retval;
+    }
+}
+
+std::vector<float> Combat::WeaponDamageImpl(std::shared_ptr<const Ship> ship, const ShipDesign* design,
+                                            float shields, bool max, bool include_fighters, bool target_ships)
+{
+    if (target_ships)
+        return WeaponDamageCalcImpl(ship, design, max, include_fighters, target_ships, CombatContextWithShipTarget(ship, shields));
+    else
+        return WeaponDamageCalcImpl(ship, design, max, include_fighters, target_ships, CombatContextWithFighterTarget(ship));
+}
+
+
+/** Populate fighter state quantities and damages for each combat round until @p limit_to_bout */
+std::map<int, Combat::FighterBoutInfo> Combat::ResolveFighterBouts(std::shared_ptr<const Ship> ship,
+                                                           const Condition::Condition* combat_targets,
+                                                   int bay_capacity, int current_docked,
+                                                   float fighter_damage, int limit_to_bout)
+{
+        // FIXME this should be moved to combat probably
+        std::map<int, FighterBoutInfo> retval;
+        const int NUM_BOUTS = GetGameRules().Get<int>("RULE_NUM_COMBAT_ROUNDS");
+        int target_bout = limit_to_bout < 1 ? NUM_BOUTS : limit_to_bout;
+
+        ScriptingContext context = CombatContextWithShipTarget(ship);
+
+        for (int bout = 1; bout <= target_bout; ++bout) {
+            context.combat_bout = bout;
+            // init current fighters
+            if (bout == 1) {
+                retval[bout].qty.docked = current_docked;
+                retval[bout].qty.attacking = 0;
+                retval[bout].qty.launched = 0;
+            } else {
+                retval[bout].qty = retval[bout - 1].qty;
+                retval[bout].qty.attacking += retval[bout].qty.launched;
+                retval[bout].qty.launched = 0;
+                retval[bout].total_damage = retval[bout - 1].total_damage;
+            }
+            // calc damage this bout, apply to total
+            int shots_this_bout = retval[bout].qty.attacking;
+            if (combat_targets && !combat_targets->Eval(context, context.effect_target)) {
+                shots_this_bout = 0;
+            }
+            retval[bout].damage = shots_this_bout * fighter_damage;
+            retval[bout].total_damage += retval[bout].damage;
+            // launch fighters
+            if (bout < NUM_BOUTS) {
+                retval[bout].qty.launched = std::min(bay_capacity, retval[bout].qty.docked);
+                retval[bout].qty.docked -= retval[bout].qty.launched;
+            }
+        }
+        return retval;
+}
+
+int Combat::TotalFighterShots(const ScriptingContext& context, const Ship& ship, const Condition::Condition* sampling_condition)
```

Comment:
indicate purpose of function and what the Condition is for in a comment probably...